hibernate-core/src/main/java/org/hibernate/event/internal/AbstractSaveEventListener.java

```diff
@@ -187,7 +188,9 @@ protected Serializable performSave(
 					throw new NonUniqueObjectException( id, persister.getEntityName() );
 				}
 			}
-			persister.setIdentifier( entity, id, source );
+			if ( ! DelayedPostInsertIdentifier.class.isInstance( id ) ) {
+				persister.setIdentifier( entity, id, source );
+			}
```

Comment:
The above is in an `if ( !useIdentityColumn ) ` block. Is it possible for `useIdentityColumn` to be `false` but `DelayedPostInsertIdentifier.class.isInstance( id )` is `true`? If so, it sounds like that would be a bug.

Reply:
You're right, this was unnecessary.  I will remove the if-check.